### PR TITLE
Change several more arrays to be assembly data spans (w/ collection expressions)

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

File: src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs [hunk 1]
```diff
@@ -1275,19 +1275,19 @@ private void AdjustCallArgumentsForParams(CType callingObjectType, CType type, M
             }
         }
 
-        private static readonly PredefinedType[] s_rgptIntOp =
-        {
+        private static ReadOnlySpan<PredefinedType> RgptIntOp() =>
```
Comment (En3Tho) [inline src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs:1278]: Any particular reason this is a method and not a property?

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Runtime/General/Helpers.cs [hunk 2]
```diff
@@ -113,7 +113,7 @@ public static string EscapeTypeNameIdentifier(this string identifier)
             // We're avoiding calling into MemoryExtensions here as it has paths that lead to reflection,
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Runtime/General/Helpers.cs:113]: This comment says that this code is intentionally avoiding `MemoryExtensions` (ie `AsSpan`). If this change passes CI, it is no longer applicable and it should be deleted.

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Runtime/General/Helpers.cs [hunk 3]
```diff
@@ -113,7 +113,7 @@ public static string EscapeTypeNameIdentifier(this string identifier)
...
             // and that would lead to an infinite loop given that this is the implementation of reflection.
 #pragma warning disable CA1870 // Use a cached 'SearchValues' instance
-            if (identifier != null && identifier.IndexOfAny(s_charsToEscape) != -1)
+            if (identifier != null && identifier.AsSpan().IndexOfAny(CharsToEscape) >= 0)
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Runtime/General/Helpers.cs:116]:
```suggestion
            if (identifier.AsSpan().Contains(CharsToEscape))
```

File: src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs:1278]: It could be a property.

Comment (En3Tho) [reply src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs:1278]: Just asking because of consistency

**Comment by stephentoub** [comment]
@jkotas, could you take another look?

**Review by jkotas — APPROVED** [review]
LGTM

**Review by jeffhandley — APPROVED** [review]
Looks good to me

**Pull request merged**